pkg/manager/member/tidb_member_manager.go

```diff
@@ -131,7 +131,7 @@ func (tmm *tidbMemberManager) syncTiDBStatefulSetForTidbCluster(tc *v1alpha1.Tid
 	tcName := tc.GetName()
 
 	newTiDBSet := tmm.getNewTiDBSetForTidbCluster(tc)
-	oldTiDBSet, err := tmm.setLister.StatefulSets(ns).Get(controller.TiDBMemberName(tcName))
+	oldTiDBSetTemp, err := tmm.setLister.StatefulSets(ns).Get(controller.TiDBMemberName(tcName))
```

Comment:
These two places need to be modified:

https://github.com/pingcap/tidb-operator/blob/b0e6cd32f9089542078f6887612be95e23b18717/pkg/manager/member/pd_member_manager.go#L175

https://github.com/pingcap/tidb-operator/blob/8bcb333e639ae4a76fc64f141acef6519931d091/pkg/manager/member/tikv_member_manager.go#L165